test/python/opflow/test_cvar.py

```diff
@@ -161,6 +163,18 @@ def test_unsupported_operations(self):
             with self.assertRaises(OpflowError):
                 cvar.adjoint()
 
+    def test_cvar_on_paulisumop(self):
+        """Test a large PauliSumOp is checked for diagonality efficiently.
+
+        Regression test for Qiskit/qiskit-terra#7573.
+        """
+        op = PauliSumOp.from_list([("Z" * 30, 1)])
+        # assert global algorithm settings do not have massive calculations turned on
+        # (which is the default, but better to be sure in the test!)
+        algorithm_globals.massive = False
+
+        _ = CVaRMeasurement(op)
```

Comment:
Is there something useful we can test on the measurement afterwards, to make sure that the object is working correctly?  Perhaps a simple call to `compute_cvar`?

I know the construction is just a proxy for calling the private method `_check_is_diagonal`, but I have a (slightly dogmatic) aversion to tests with no explicit assertions, because they can easily lose meaning if the verification paths change in the future.

Reply:
haha I was thinking someone might object to that! I added a check statement using `compute_cvar`.